### ISSUE TITLE
Import is labeled as error E0401 when file is being also imported

### ISSUE THREAD

**Issue by JavierReyes945** [PyCQA/pylint#2619]
False positive when importing a module from a script that is also imported.

Might be related to #2603 (but in this case it is related to user scripts).

### Steps to reproduce

Folder structure:

```
src
|--- main.py
|--- folderA
      |--- module1
      |--- module2
      |--- module3
|--- folderB
      |--- ...
```

1. `main.py` imports a class with: `from folderA.module1 import module1`
2. `module1` inherits from `module3` with: `from folderA.module3 import module3`.

At step 2, `sys.path` is still pointing at the directory `src`.

### Current behavior

Pylint labels the statement in step 2 with error `[pylint] Unable to import 'features.Parser' [E0401]`.

### Expected behavior

Script `main.py` executed directly works as expected, modules are correctly imported.

### pylint --version output

```
> pylint --version
pylint 2.1.1
astroid 2.0.4
Python 3.7.1 (default, Oct 28 2018, 08:39:03) [MSC v.1912 64 bit (AMD64)]
```

**Comment by PCManticore** [comment]
Hey @JavierReyes945 Thanks for reporting an issue? would you mind writing this reproduction case as a script that creates everything or if you have a minimal repository instead, that would be super helpful. As it is right now, it's kind of hard to figure out what and where the problem is. Thanks.
